plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHivePartitionedBucketFunction.java

```diff
@@ -124,6 +126,29 @@ public void testMultiplePartitions(BucketingVersion hiveBucketingVersion)
                 numBuckets * numPartitions);
     }
 
+    @Test(dataProvider = "hiveBucketingVersion")
+    public void testConsecutiveBucketsWithinPartition(BucketingVersion hiveBucketingVersion)
+    {
+        BlockBuilder bucketColumn = BIGINT.createFixedSizeBlockBuilder(10);
+        BlockBuilder partitionColumn = BIGINT.createFixedSizeBlockBuilder(10);
+        for (int i = 0; i < 100; ++i) {
+            bucketColumn.writeLong(i);
+            partitionColumn.writeLong(42);
+        }
+        Page page = new Page(bucketColumn, partitionColumn);
+
+        BucketFunction hivePartitionedBucketFunction = partitionedBucketFunction(hiveBucketingVersion, 10, ImmutableList.of(HIVE_LONG), ImmutableList.of(BIGINT), 4000);
+        List<Integer> positions = new ArrayList<>();
+        for (int i = 0; i < 100; ++i) {
+            positions.add(hivePartitionedBucketFunction.getBucket(page, i));
+        }
+
+        int minPosition = min(positions);
+        int maxPosition = max(positions);
+
+        assertEquals(maxPosition - minPosition + 1, 10);
```

Comment:
Why 10?

Reply:
Please also extract 10 as `hiveBucketCount`. Without that it is still cryptic IMO.